From 6efcc17cb8ff4660ccf586f3028665d976f17ecb Mon Sep 17 00:00:00 2001
From: Jouni Malinen <quic_jouni@quicinc.com>
Date: Thu, 10 Oct 2024 23:45:07 +0300
Subject: [PATCH] P2P: Clear GO negotiation results from stack after use

struct p2p_go_neg_results contains private keys, so clear it from stack
explicitly to avoid leaving any unnecessary copies of keys in memory.

Signed-off-by: Jouni Malinen <quic_jouni@quicinc.com>
---
 src/p2p/p2p.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/p2p/p2p.c b/src/p2p/p2p.c
index 907cc08a8..13e30231e 100644
--- a/src/p2p/p2p.c
+++ b/src/p2p/p2p.c
@@ -1931,6 +1931,7 @@ void p2p_go_complete(struct p2p_data *p2p, struct p2p_device *peer)
 
 	p2p_set_state(p2p, P2P_PROVISIONING);
 	p2p->cfg->go_neg_completed(p2p->cfg->cb_ctx, &res);
+	forced_memzero(&res, sizeof(res));
 }
 
 
-- 
2.43.0

